presto-memory-context/src/main/java/com/facebook/presto/memory/context/SimpleLocalMemoryContext.java

```diff
@@ -87,12 +91,12 @@ public synchronized void transferOwnership(LocalMemoryContext to)
             parent = parent.parentMemoryContext;
         }
 
-        // If parent is null at this point, we fail as the memory context state is probably corrupt.
-        // Also, there is also a possibility of a deadlock if we continue in that case.
+        // if parent is null at this point, we fail as the memory context state is probably corrupt.
+        // there is also a possibility of a deadlock if we continue in that case.
         checkState(parent != null, "memory context state is corrupt");
 
         synchronized (target) {
-            target.usedBytes += usedBytes;
+            target.usedBytes = AggregatedMemoryContext.addExact(target.usedBytes, usedBytes);
```

Comment:
Please do make `usedBytes` and expose "backdoor method" for this. Like `directSetBytes(long bytes)`